parse_test.go

```diff
@@ -860,6 +860,53 @@ func TestEnvironmentVariableInSubcommandIgnored(t *testing.T) {
 	assert.Equal(t, "", args.Sub.Foo)
 }
 
+func TestParserMustParseEmptyArgs(t *testing.T) {
+	// this mirrors TestEmptyArgs
+	p, err := NewParser(Config{}, &struct{}{})
+	require.NoError(t, err)
+	assert.NotNil(t, p)
+	p.MustParse(nil)
+}
+
+func TestParserMustParse(t *testing.T) {
+	tests := []struct {
+		name    string
+		args    versioned
+		cmdLine []string
+		code    int
+		output  string
+	}{
+		{name: "help", args: struct{}{}, cmdLine: []string{"--help"}, code: 0, output: "display this help and exit"},
+		{name: "version", args: versioned{}, cmdLine: []string{"--version"}, code: 0, output: "example 3.2.1"},
+		{name: "invalid", args: struct{}{}, cmdLine: []string{"invalid"}, code: -1, output: ""},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			originalExit := osExit
+			originalStdout := stdout
+			defer func() {
+				osExit = originalExit
+				stdout = originalStdout
+			}()
+
+			var exitCode int
```

Comment:
initialize this to `-1` so that when the expected exit code is zero, the `assert.Equals` line below really checks that osExit was called.

Reply:
If it's -1 then it wouldn't help us be certain about the invalid case. I'll swap this back to being an int pointer, then we can differentiate between 0, nil etc.